svc-events/events/publishevttosubscriber.go

```diff
@@ -110,41 +64,50 @@ func PublishEventsToDestination(data interface{}) bool {
 
 	var flag bool
 	var uuid string
-
-	checkAndAddFabrics(requestData, host)
+	var message common.MessageData
 
 	deviceSubscription, err := evmodel.GetDeviceSubscriptions(host)
 	if err != nil {
-		log.Printf("error: Failed to get the event destinations: %v\n", err)
+		log.Info("error: Failed to get the event destinations: ", err.Error())
 		return false
 	}
 
 	if len(deviceSubscription.OriginResources) < 1 {
-		log.Println("error: no origin resources found in device subscriptions")
+		log.Info("error: no origin resources found in device subscriptions")
 		return false
 	}
 
 	requestData, uuid = formatEvent(requestData, deviceSubscription.OriginResources[0], host)
-	message, parseStatus := parseEventData(requestData)
-	if !parseStatus {
-		log.Println("Error while trying to parse the input request data", message)
+
+	subscriptions, err := evmodel.GetEvtSubscriptions(host)
+	if err != nil {
 		return false
 	}
 
-	subscriptions, err := evmodel.GetEvtSubscriptions(host)
+	err = json.Unmarshal([]byte(requestData), &message)
 	if err != nil {
+		log.Info("error: failed to unmarshal the incoming event:", requestData, "with the error:", err.Error())
 		return false
 	}
 
-	eventMap := make(map[string][]ForwardEvent)
+	eventMap := make(map[string][]common.Event)
 	for _, inEvent := range message.Events {
-		if len(inEvent.OriginOfCondition) < 1 {
-			log.Printf("event not forwarded : Originofcondition is empty in incoming event with body %v\n", requestData)
+		if inEvent.OriginOfCondition == nil {
+			log.Info("event not forwarded : Originofcondition is empty in incoming event with body: ", requestData)
+			continue
+		}
+
+		if len(inEvent.OriginOfCondition.Oid) < 1 {
+			log.Info("event not forwarded : Originofcondition is empty in incoming event with body: ", requestData)
 			continue
 		}
+		if strings.EqualFold(inEvent.EventType, "ResourceAdded") &&
+			strings.HasPrefix(inEvent.OriginOfCondition.Oid, "/redfish/v1/Fabrics") {
+			addFabricRPCCall(inEvent.OriginOfCondition.Oid, host)
```

Comment:
Please cross check with CFM plugin, earlier addFabricRPCCall was not getting called when an fabric added events comes to odimra

Reply:
I have tested with event-publisher binary by creating fabric event, `addFabricRPCCall ` method getting called.